addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerDetailsEdit.js

```diff
@@ -19,13 +19,8 @@ export class PartnerDetailsEdit extends Component {
             state_id: partner.state_id && partner.state_id[0],
             property_product_pricelist: this.setDefaultPricelist(partner),
         };
-
-        onMounted(() => {
-            this.env.bus.on("save-partner", this, this.saveChanges);
-        });
-
-        onWillUnmount(() => {
-            this.env.bus.off("save-partner", this);
+        Object.assign(this.props.imperativeHandle, {
+            save: () => this.saveChanges(),
```

Comment:
This refactor on partner screen is out of place. It probably deserves a separate commit.

Reply:
The reason it's in this commit is that because we stop using the legacy environment, it no longer contains a bus, causing the previous code to crash. I don't think this diff is substantial enough to merit its own commit but if you think so I'll extract it out.

Reply:
Oh I see. It's fine to keep it here. No need to extract.